app/src/ui/rename-branch/rename-branch-dialog.tsx

```diff
@@ -31,10 +33,13 @@ export class RenameBranch extends React.Component<
   public constructor(props: IRenameBranchProps) {
     super(props)
 
-    this.state = { newName: props.branch.name }
+    this.state = { newName: props.branch.name, renaming: false }
   }
 
   public render() {
+    const icon = this.state.renaming ? (
```

Comment:
Rather than including an icon in the button here we should use the new state that you've added and set the `loading` prop of the `<Dialog>` component instead. See https://github.com/desktop/desktop/blob/master/app/src/ui/add-repository/create-repository.tsx#L459 for an example.

Setting the loading prop of the Dialog will automatically disable all buttons in the dialog and show a spinner in the dialog header. By having a uniform way of signalling that a dialog is busy we can more easily change the behavior in the future should we wish to do so.